### DISCUSSION TITLE
What is the "correct" way to convert WAS BOOLEANs into numbers to switch other nodes?

### DISCUSSION THREAD

**Discussion by curiousjp** [WASasquatch/was-node-suite-comfyui#349] [Q&A]
I am interested if anyone has a more elegant solution to this problem. Say I have two strings, and I want to switch between them depending on whether one of them contains a certain control sequence. _Text Contains_ provides a method for detecting the sequence and provides a BOOLEAN in response, but _Text Input Switch_ requires a boolean number instead of a boolean. (Similarly, nodes like ImpactPack's _Switch (Any)_ want an index number.)

As far as I can tell, the approach is to use _Boolean to Text_, then use that text representation as a key for _Text Dictionary Get_ to retrieve an index from a dictionary as a string, then use _Text to Number_, and then use this for something like _Text Input Switch_. Is there a better, more direct method, for converting BOOLEAN to an index/boolean number?

![workflow](https://github.com/WASasquatch/was-node-suite-comfyui/assets/48515264/971a4273-3f8c-409c-9870-6dc25d98ddee)

**Comment by WASasquatch** [comment]
I thought the switches were converted to use normal bools? I may have
missed some. It should accept normal book output of the other nodes (which
shouldn't be Number now).

**Comment by curiousjp** [comment]
> I may have missed some. It should accept normal book output of the other nodes (which shouldn't be Number now).

I think this one might have been overlooked - if you have a look at line [13096](https://github.com/WASasquatch/was-node-suite-comfyui/blob/main/WAS_Node_Suite.py?plain=1#L13096) of WAS_Node_Suite.py, it still lists "boolean_number": ("NUMBER",) as a required input to WAS_Text_Input_Switch. WAS_Text_Contains on line [13131](https://github.com/WASasquatch/was-node-suite-comfyui/blob/main/WAS_Node_Suite.py?plain=1#L13131) still has a return type of ("BOOLEAN",).

**Comment by adrian-forest** [comment]
Is this issue related? #364
